target/linux/ramips/dts/DWR-922-E2.dts

```diff
@@ -0,0 +1,152 @@
+/dts-v1/;
+
+#include "mt7620n.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "dlink,dwr-922-e2", "ralink,mt7620n-soc";
+	model = "D-Link DWR-922 E2";
+
+	aliases {
+		led-boot = &led_sstrenghg;
+		led-failsafe = &led_sstrenghg;
+	};
+
+	gpio-keys-polled {
+		compatible = "gpio-keys-polled";
+		poll-interval = <20>;
+
+		wps {
+			label = "wps";
+			gpios = <&gpio0 2 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+
+		reset {
+			label = "reset";
+			gpios = <&gpio0 1 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+
+	gpio-leds {
+		compatible = "gpio-leds";
+
+		sms {
+			label = "dwr-922-e2:green:sms";
+			gpios = <&gpio1 14 GPIO_ACTIVE_LOW>;
+		};
+
+		lan {
+			label = "dwr-922-e2:green:lan";
+			gpios = <&gpio1 15 GPIO_ACTIVE_HIGH>;
+		};
+
+		led_sstrenghg: sstrengthg {
```

Comment:
The label should follow upstream Documentation/leds/leds-class.txt's "LED Device Naming"
recommendations: "devicename:colour:function" and I doubt, that `sstrengthg` is function. Could you please come with better names and aliases? Maybe `modem-link` or `modem-activity` would work better and wouldn't be that cryptic?

Reply:
`devicename:colour:function`, you've `colour` already so you don't need to put color into `function` again, so maybe `rssi-low`, `rssi-medium`, `rssi-high` would work for `function`?

Reply:
@squigley If get you back to before the attempt at squashing and you don't try the 'shortcut' @ynezz suggested but instead do plain squashing, are you willing to try again?

It can be a little tricky, so here is a major caveat:

git rebase -i commit^ should point to the commit you want to change **but** watch the messages...if there is an error in rebase you can end up on the __previous__ commit (which is what I think happened to you) and then things get quite messed up.

If you're rebasing more than one commit this can happen anywhere in the set of commits you are operating on.

Also, my preferred way of syncing with master (I think it's safer) is this:

1. ``git checkout master``
2. ``git checkout -b pr-new-branch``
3. `gitk pr-new-branch pr-old-branch`` and right-click to cherry pick (visually) the commits from your pr-old-branch branch (i.e. what you had originally) OR (for each commit to move) ``git cherry-pick <commit-in-pr-old-branch>`` (since your on pr-new-branch that will cherry-pick <commit-in-pr-old-branch> to pr-new-branch).`` **NB** you may get a message about a conflict, which you have to resolve like you would a merge conflict, then ``git cherry-pick --continue``
4. **Verify** all commits you want are in pr-new-branch and there is nothing left you want in pr-old-branch (if that is not the case, ``git checkout pr-old-branch`` and then ``git checkout -b pr-old-branch-keep`` and then ``git checkout pr-new-branch`` to continue as before)
5. ``git  branch -D pr-old-branch``
6. ``git checkout -b pr-old-branch``
7. THEN force-push (e.g. ``git push origin +pr-old-branch``)

Reply:
OK, I tried this, but it was still confusing and didn't seem to do what I needed to do. I did a "git reset --hard HEAD" and made all my changes again, added, committed, pushed, and made a new PR. https://github.com/openwrt/openwrt/pull/1660

Reply:
In response to the start of this conversation, regarding the LEDs, the documentation for the device has:

Signal Strength
Blinking Red: No SIM card / signal or unverified PIN code
Solid Red: Signal strength is at level one (weak)
Solid Amber: Signal strength is at level two or three (medium)
Solid Green: Signal strength is at level four or five (strong)

It's a bi-colour LED, and I looked closely while powering it up as it is orange at that point, and I see that it is just activating both red and green at the same time.